dataland-frontend/src/components/forms/parts/UploadReports.vue

```diff
@@ -147,42 +160,60 @@ export default defineComponent({
       }
     },
     /**
-     * Handles selection of a file by the user. If duplicates are found in the selection, this is handled.
+     * Handles selection of files by the user. If invalid file names are found in the selection, this is handled.
      * At the end an event is emitted reflecting that the referenceable reports have updated.
-     * @param documents the list of all documents for the upload, determined by the selection in the file uploader
+     * @param selectedDocumentsForUpload the list of all selectedDocumentsForUpload for the upload, determined by the selection in the file uploader
      */
-    handleUpdatedDocumentsSelectedForUpload(documents: DocumentToUpload[]) {
-      this.documentsToUpload = documents;
-      const duplicatesWithIndex: DuplicateWithIndex[] = [];
+    handleUpdatedDocumentsSelectedForUpload(selectedDocumentsForUpload: DocumentToUpload[]) {
+      this.documentsToUpload = selectedDocumentsForUpload;
 
-      if (this.areDuplicatesAmongReferenceableReportNames()) {
-        const foundExistingRecords = new Set<string>();
+      const invalidFileNamesWithIndexAndReason: InvalidFileNameWithIndexAndReason[] = [];
+      const existingFileNamesCollector = new Set<string>();
 
-        for (let i = 0; i < this.documentsToUpload.length; i++) {
-          const currentName = this.documentsToUpload[i].fileNameWithoutSuffix;
+      for (let i = 0; i < this.documentsToUpload.length; i++) {
+        const fileName = this.documentsToUpload[i].fileNameWithoutSuffix;
 
-          if (foundExistingRecords.has(currentName) || this.namesOfStoredReports.indexOf(currentName) !== -1) {
-            duplicatesWithIndex.push({ document: this.documentsToUpload[i], index: i });
-          } else {
-            foundExistingRecords.add(currentName);
-          }
+        if (this.hasFileNameForbiddenCharacter(fileName)) {
+          invalidFileNamesWithIndexAndReason.push({
+            fileName: fileName,
+            index: i,
+            invalidityReason: FileNameInvalidityReason.ForbiddenCharacter,
+          });
+        } else if (existingFileNamesCollector.has(fileName) || this.namesOfStoredReports.indexOf(fileName) !== -1) {
+          invalidFileNamesWithIndexAndReason.push({
+            fileName: fileName,
+            index: i,
+            invalidityReason: FileNameInvalidityReason.Duplicate,
+          });
+        } else {
+          existingFileNamesCollector.add(fileName);
         }
+      }
 
-        this.handleReportDuplicates([...duplicatesWithIndex].reverse());
+      if (invalidFileNamesWithIndexAndReason.length > 0) {
+        this.handleInvalidFileNames([...invalidFileNamesWithIndexAndReason].reverse());
```

Comment:
The name `handleInvalidFiles` would fit. This one does not.

Reply:
How about `handleFilesWithInvalidNames()` ?
Because the files themselves (meaning their content) is not detected as invalid here. It is just the names that are not ok.

Reply:
👍 fair

Reply:
Renamed